app/services/cfe/submission_manager.rb

```diff
@@ -46,15 +46,24 @@ def submission
     end
 
     def call_common_services
-      COMMON_SERVICES.each { |s| s.call(submission) }
+      COMMON_SERVICES.each do |service|
+        log_duration(service)
+      end
     end
 
     def call_non_passported_services
       return if submission.passported?
 
-      NON_PASSPORTED_SERVICES.each do |s|
-        s.call(submission)
+      NON_PASSPORTED_SERVICES.each do |service|
+        log_duration(service)
       end
     end
+
+    def log_duration(service)
```

Comment:
might be worth making reusable and use a block

for example, not test by any means

```
def log_duration(&block)
  start_time
  yield block if block_given?
  Rails.logger.info("duration: #{end_time - end_time}...blah")
  end_time
end
```

```
NON_PASSPORTED_SERVICES.each do |service|
        log_duration do
          service.call(submission)
      end
end
```

Reply:
**not a blocker**